dockerfiles/Dockerfile.ubuntu12.from-tar

```diff
@@ -24,14 +24,21 @@ RUN apt-get update && \
             texinfo
 
 # Build emacs
-ARG GIT_REPOSITORY="git://git.sv.gnu.org/emacs.git"
+ARG GIT_REPOSITORY="http://git.sv.gnu.org/cgit/emacs.git/snapshot"
 ARG GIT_BRANCH
-RUN git clone --depth 1 --branch $GIT_BRANCH $GIT_REPOSITORY /tmp/emacs && \
-    cd /tmp/emacs && \
-    ./autogen.sh && \
-    ./configure && \
+
+# Tagged releases are available as tar balls, and should be smaller
+# and faster to download.  The build steps below are specifically for
+# Emacs 23, not correct for 24+.
+
```

Comment:
Hum, if we use the tagged release over HTTP then we'd rename the `GIT_*` variables too.
I'm not sure we really care about the source clone speed tho, this is all done in travis-ci and the clone/download time are likely small compared to the emacs building time.

Reply:
At one point I found Emacs developer docs stating their release numbering conventions. I can't find it right now, but ....

Only the major and minor number is considered meaningful ("25.3").  If there is a third "build" number (as in "26.0.91") then that indicates a development build and not a true release. There are ad hoc conventions how they use the third number in their development builds but there's no guarantee those conventions will be strictly followed.

If there is a major flaw or security issue then what I see them do is release a new version (25.3 was such a case, a CVE was reported on 25.2 and earlier). You can see the timeline of tags in their git repo is strictly ordered too.

Reply:
About using the tar files and maybe changing the `GIT_*` variables, let me do some more experimentation on the sequential builds... if I'm asking Travis CI to cache the source trees anyway then this optimization isn't really necessary.